src/loader.js

```diff
@@ -210,18 +212,42 @@ async function loader(content) {
       ? this.resourcePath
       : path.relative(this.rootContext, this.resourcePath);
 
-  const minifyOptions =
-    /** @type {import("./index").InternalWorkerOptions<T>} */ ({
-      input: content,
-      filename,
-      severityError,
-      transformer,
-      generateFilename:
-        /** @type {Compilation} */
-        (this._compilation).getAssetPath.bind(this._compilation),
-    });
-
-  const output = await worker(minifyOptions);
+  if (!this._compilation || !this._compiler) {
+    callback(new Error("_compilation and/or _compiler unavailable"));
+    return;
+  }
+
+  const logger = this._compilation.getLogger("ImageMinimizerPlugin");
+  const cache = this._compilation.getCache("ImageMinimizerWebpackPlugin");
+
+  const { RawSource } = this._compiler.webpack.sources;
+  const eTag = cache.getLazyHashedEtag(new RawSource(content));
+  const cacheName = getSerializeJavascript()({
+    eTag: eTag.toString(),
+    transformer: (Array.isArray(transformer) ? transformer : [transformer]).map(
+      (each) => ({
+        implementation: each.implementation,
+        options: each.options,
+      }),
+    ),
+  });
+  const cacheItem = cache.getItemCache(cacheName, eTag);
+  const output = await cacheItem.providePromise(() => {
+    const minifyOptions =
+      /** @type {import("./index").InternalWorkerOptions<T>} */ ({
+        input: content,
+        filename,
+        severityError,
+        transformer,
+        generateFilename:
+          /** @type {Compilation} */
+          (this._compilation).getAssetPath.bind(this._compilation),
+      });
+
+    logger.debug(`loader cache miss: ${filename}`);
+
+    return worker(minifyOptions);
+  });
```

Comment:
We can't do it in loader:
1. `this._compilation.getCache` is unavaliable in multi threading mode (`thread-loader`)
2. Also this plugin is using by rspack too and such api is not working there 


> This PR eliminates the race condition where the loader or optimizer may do the same processing (same transformation on the same image) redundantly if the same file transformation's loader is triggered more than once before the first transformation has finished.

Can you provide reproducible example?

Also I think we can use `WeakMap` in the loader to prevent such situations